commit 9edbf5e03dd818a1cd3acd684a25206bd5a2cfd2 (refs/changes/10/2810/2)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-10-18T21:00:48+02:00 (2 years ago)
    
    TOOLS-1897 sdcadm experimental update-other should update application domains once
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Julien Gilli <julien.gilli@joyent.com>

diff --git a/lib/cli/do_update_other.js b/lib/cli/do_update_other.js
index d8f55b7..b09f1df 100644
--- a/lib/cli/do_update_other.js
+++ b/lib/cli/do_update_other.js
@@ -286,15 +286,9 @@ function do_update_other(subcmd, opts, args, cb) {
             if (Object.keys(mdataUpdates).length === 0) {
                 return next();
             }
-
             progress('Adding domain keys to "sdc" SAPI app metadata: %j',
                 mdataUpdates);
-            vasync.forEachParallel({
-                inputs: ctx.svcs,
-                func: function updateApp(svc, nextSvc) {
-                    updateSdcApp({metadata: mdataUpdates}, nextSvc);
-                }
-            }, next);
+            updateSdcApp({metadata: mdataUpdates}, next);
         },
 
         function getSdc4096Pkg(ctx, next) {
diff --git a/lib/steps/sapi.js b/lib/steps/sapi.js
index bf7c6f3..f17e4a1 100644
--- a/lib/steps/sapi.js
+++ b/lib/steps/sapi.js
@@ -150,7 +150,7 @@ function sapiEnsureAgentServices(arg, cb) {
                 return next(new errors.SDCClientError(new Error('Datacenter ' +
                     'does not have the minimum SAPI version needed for adding' +
                     ' service agents. ' +
-                    'Please try again after upgrading SAPI')));
+                    'Please try again after upgrading SAPI', 'sapi')));
             }
 
             return next();
